### ISSUE TITLE
API consistency checks don't check that protected methods are virtual

### ISSUE THREAD

**Issue by ajcvickers** [aspnet/EntityFramework#1073]
This is related to #462, but filing as a separate issue since that issue is both blocked by a VS bug and also is about NotNull attributes, which have potentially less value on protected methods than virtual which is often very important for protected methods. So I think we should fix this even if not addressing the NotNull attribute issue.
